src/Storages/StorageDictionary.h

```diff
@@ -31,13 +31,21 @@ class StorageDictionary final : public ext::shared_ptr_helper<StorageDictionary>
     const String & dictionaryName() const { return dictionary_name; }
 
 private:
-    String dictionary_name;
+    const String dictionary_name;
+    const bool internal = false;
 
 protected:
     StorageDictionary(
         const StorageID & table_id_,
         const String & dictionary_name_,
-        const DictionaryStructure & dictionary_structure);
+        const ColumnsDescription & columns_,
+        bool internal_ = false);
```

Comment:
Missing comment.
(I don't understand what does it mean.)

Maybe remove default value (false) and always specify it explicitly?